app/models/notification/purging.rb

```diff
@@ -0,0 +1,68 @@
+class Notification
+  module Purging
+    extend ActiveSupport::Concern
+    include PurgingMixin
+
+    module ClassMethods
+      def purge_date
+        nil
+      end
+
+      def purge_window_size
+        ::Settings.notifications.history.purge_window_size
+      end
+
+      # Collects the notifications that have expired based on the expires_in
+      # value for each notification type.  The "older_than" value is not used
+      # for this function, so it is named as `_`.
+      #
+      # First, this collects all of the notification_types and groups them by
+      # expires_in value (we only seed 3 different types currently).
+      #
+      # It then builds a single query that matches against each
+      # notification_type and if it has expired that specific expires_in for
+      # that type.
+      def purge_scope(_)
+        query = nil
+
+        # Make a query to collect the expires_in and id columns for all of the
+        # NotificationTypes, and then put them into a hash with the key being
+        # the expires_in column, and the values being all of the ids matching
+        # the expires_in value.
+        expire_ids = NotificationType.pluck(:expires_in, :id)
+        expire_ids = expire_ids.each_with_object({}) do |(key, id), result|
+          result[key] ||= []
+          result[key] << id
+        end
```

Comment:
```
NotificationType.pluck(:expires_in, :id).group_by(&:first).each { |k, v| v.replace(v.transpose.last) }
```

Maybe?

Not even sure you need to do the transpose bit here, because you can always just do that later as you loop over the grouped things.

Reply:
Oh no wait....this one is much more fun

```ruby
NotificationType.pluck(:expires_in, :id).group_by(&:shift).each_value(&:flatten!)
```

Reply:
Yeah, I wasn't sold on what I did here, I just got something that worked.  I can try these out on Monday.  Thanks for the suggestions.

Reply:
The above suggestion might now not be needed based on the query suggestions below (basically replaces this entire method), but will keep this in mind if the more complex query ends up being more performant on the DB (again, still need to test at some form of scale).